tycho-extras/tycho-pomless/src/main/java/org/eclipse/tycho/pomless/TychoFeatureMapping.java

```diff
@@ -54,7 +65,7 @@ protected void initModelFromXML(Model model, Element xml, File artifactFile) thr
 
     @Override
     protected boolean isValidLocation(String location) {
-        return location.endsWith(FEATURE_XML);
+        return Path.of(location).endsWith(FEATURE_XML);
```

Comment:
Is there any advantage of constructing a path first? Could this be a separate PR?

Reply:
This makes the test more precise. Previously that mapping would also handle a project that contains a file named:
`someOtherStringfeature.xml`

Reply:
But then we need some error handling as the location not necessarily must be a java path (even though it is very likely)

Reply:
If we do that, it should be consistent with all the other places that create Files without error handling, for example `TychoBundleMapping.isValidLocation()` and many more.
Furthermore I also wonder how errors would ideally be handled? Fallback to the String.endsWith() in a catch-block? Or just throw an Exception? Because as you said flawed paths are unlikely we maybe would better be told when they occur?

However according to the Java-doc new File() only throws an NPE, while Path.of() can throw a invalid Path exception.
So following that `new File(location).getName().equals(FEATURE_XML)`, should not throw any exception.

Reply:
Correct, `new File()` never throws any exception (except fro null) but `Path.of()` does when it think it is invalid. This can lead to strange results, but as we never know what client code passes in as a location one should better be safe here, making it fail to parse even though might simply result in false being returned.

Reply:
To handled this consistently across all Mappings this requires changes at multiple locations and migration from the `File` to the `Path` API, to not switch back and forth.
I extracted those changes to a third commit.